lib/phoenix_mtm/helpers.ex

```diff
@@ -30,14 +29,73 @@ defmodule PhoenixMTM.Helpers do
     * `:selected` - a list of options that should be pre-selected
     * `:input_opts` - a list of attributes to be applied to each checkbox input
     * `:label_opts` - a list of attributes to be applied to each checkbox label
+    * `:wrapper` - a function to wrap the HTML structure of each checkbox/label
+    * `:mapper` - a function to customize the HTML structure of each checkbox/label
+
+
+  ## Wrapper
+
+  A `wrapper` function can be used to wrap each checkbox and label pair in one
+  or more HTML elements.
+
+  The wrapper function receives the pair as a single argument, and should return
+  a `safe` tuple as expected by Phoenix.
+
+  A simplified version of this is to call `Phoenix.HTML.Tag.content_tag`
+
+      <%= PhoenixMTM.Helpers.collection_checkboxes f, :tags,
+            Enum.map(@tags, &({&1.name, &1.id})),
+            selected: Enum.map(f.data.tags, &(&1.id)),
+            wrapper: &Phoenix.HTML.Tag.content_tag(:p, &1)
+
+
+  ## Mapper
+
+  A `mapper` function can be used to customize the structure of the checkbox and
+  label pair.
+
+  The mapper function receives the form, field name, input options, label text,
+  label options, and helper options, and should return a `safe` tuple as expected
+  by Phoenix.
+
+      # Somewhere in your application
+      defmodule CustomMappers do
+        use PhoenixMTM.Mappers
+
+        def bootstrap(form, field, input_opts, label_text, label_opts, _opts) do
+          content_tag(:div, class: "checkbox") do
+            label(form, field, label_opts) do
+              [
+                tag(:input, input_opts),
+                {:safe, "#{label_text}"}
+              ]
+            end
+          end
+        end
+      end
+
+      # In your template
+      <%= PhoenixMTM.Helpers.collection_checkboxes f, :tags,
+            Enum.map(@tags, &({&1.name, &1.id})),
+            selected: Enum.map(f.data.tags, &(&1.id)),
+            mapper: &CustomMappers.bootstrap/6
   """
   def collection_checkboxes(form, field, collection, opts \\ []) do
     name = field_name(form, field) <> "[]"
     selected = Keyword.get(opts, :selected, [])
     input_opts = Keyword.get(opts, :input_opts, [])
     label_opts = Keyword.get(opts, :label_opts, [])
+    mapper = Keyword.get(opts, :mapper, &PhoenixMTM.Mappers.unwrapped/6)
+    wrapper = Keyword.get(opts, :wrapper, &(&1))
+
+    # TODO: Eventually deprecate this option in favour of passing in custom mapper
+    mapper = if {:nested, true} in opts do
+      &PhoenixMTM.Mappers.nested/6
+    else
+      mapper
+    end
 
-    inputs = Enum.map(collection, fn {label, value} ->
+    inputs = Enum.map(collection, fn {label_text, value} ->
```

Comment:
If this is changed to `label_value`, rather than `label_text`, then we make it clear that it also supports cases where this can contain rich data to be used in a `mapper`, rather than just text.

One concrete use case I have is as follows:
I want to create rich labels based on parameters of an object. For example if the label object has a certain property, I want to generate some extra HTML (like a bootstrap label for example). This only works if it is an object, rather than just a text string. Generating these rich values (containing html) prior to calling the PhoenixMTM helper is clumsy since it clashes with escaping the HTML values.